### PR TITLE
feat(git): add new entry(rebase) in git subdirectory

### LINKED ISSUE #3490: [Concept Entry] Git Rebase
### Reason/inspiration (optional)

We would like a new concept entry for Git about Rebase. The entry should go in a new file in docs/content/git/concepts/rebase/rebase.md

The entry should
    - Have an introductory paragraph about the concept.
    - Code blocks displaying different aspects of the concept.
    - Internal linking as applicable.

Please refer to the [concept entry templates](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by adeyinkaezra123**
### Description
Add new [Concept Entry] Git Rebase. Fixes #3490 

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/git/concepts/rebase/rebase.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+4. Start the rebase: Start a rebase in either the regular or interactive modes
+
+5. Resolve conflicts (if any): Git will pause the rebase if there are conflicts. Use git status to identify the files with conflicts and resolve them manually. After resolving conflicts, use `git add <file>` and `git rebase --continue` to continue the rebase.
+
+6. Complete the rebase: After resolving conflicts (if any), the rebase will continue automatically or when you use `git rebase --continue`.
+
+7. Push the rebased branch: If you're working on a shared branch, it's essential to push the rebased branch with `git push` to update the remote branch.
+
+8. Clean up (optional): Once you're sure the rebase is successful, you can delete the old branch if needed with `git branch -d <old-branch-name>`.
```
Comment (SSwiniarski) [inline content/git/concepts/rebase/rebase.md:52]:
```suggestion
1. Make sure the working directory is clean: Commit or stash any changes in the current branch to ensure there are no uncommitted modifications.
2. Checkout the branch to rebase: Use `git checkout <branch-name>` to switch to the branch to rebase.
3. Fetch the latest changes: Run `git fetch` to capture the latest changes from the remote repository.
4. Start the rebase: Start a rebase in either the regular or interactive modes
5. Resolve conflicts (if any): Git will pause the rebase if there are conflicts. Use git status to identify the files with conflicts and resolve them manually. After resolving conflicts, use `git add <file>` and `git rebase --continue` to continue the rebase.
6. Complete the rebase: After resolving conflicts (if any), the rebase will continue automatically or when `git rebase --continue` is used.
7. Push the rebased branch: When working on a shared branch, it's essential to push the rebased branch with `git push` to update the remote branch.
8. Clean up (optional): Once the rebase is successful, the old branch can be deleted if needed with `git branch -d <old-branch-name>`.
```
There shouldn't be blank lines between elements of a bulleted list.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@adeyinkaezra123 I did a first review and left a few comments.

**Comment by adeyinkaezra123** [comment]
@SSwiniarski Incorporated all suggestions

**Review by SSwiniarski — APPROVED** [review]
@adeyinkaezra123 I fixed one typo. Looks good for a second review.

**Review by caupolicandiaz — APPROVED** [review]
@adeyinkaezra123 @SSwiniarski LGTM

**Pull request merged**